x-pack/plugins/apm/server/lib/services/get_throughput.ts

```diff
@@ -49,30 +52,27 @@ export async function getThroughput({
 }: Options) {
   const { apmEventClient } = setup;
 
-  const filter: ESFilter[] = [
-    { term: { [SERVICE_NAME]: serviceName } },
-    { term: { [TRANSACTION_TYPE]: transactionType } },
-    ...getDocumentTypeFilterForTransactions(searchAggregatedTransactions),
-    ...rangeQuery(start, end),
-    ...environmentQuery(environment),
-    ...kqlQuery(kuery),
-  ];
-
-  if (transactionName) {
-    filter.push({
-      term: {
-        [TRANSACTION_NAME]: transactionName,
-      },
-    });
-  }
-
   const params = {
     apm: {
       events: [getProcessorEventForTransactions(searchAggregatedTransactions)],
     },
     body: {
       size: 0,
-      query: { bool: { filter } },
+      query: {
+        bool: {
+          filter: [
+            { term: { [SERVICE_NAME]: serviceName } },
```

Comment:
I'm a bit on the fence about required params. I think it's pretty easy to read already so a custom dsl might do more harm than good here. However, since we are already using `termQuery` elsewhere it might be less confusing to simply use it everywhere (and not just for optional params).

Reply:
... thinking about this a bit more, I tend to agree, that we should probably just use `termQuery` consistently everywhere.